### PR TITLE
#324 dirty mSupply cludge fix

### LINKED ISSUE #324: Unified codes lookup returning entities at incorrect level
See original mSupply issue: https://github.com/sussol/msupply/issues/8211.

## Expected Behaviour

mSupply query should return entities at the old `unit_of_use` level, i.e. `DoseStrength`.

## Current Behaviour

mSupply query is returning only `Product` entities.

## Possible Solution

Due to time constraints and updating mSupply deployments not being an option, this is going to be messy 😢 . Quick fix is to patch in some conditional logic for mSupply queries (identified by isolating types/fields which the UI does not currently query).

Note: longer term, we need to address the inconsistencies between how mSupply and the web UI use the graph exposed by the `data-service` (in particular, how they conceptualise an entity type and how this maps to the concept of entity levels).

## Steps to Reproduce (bugs only, delete if not applicable)

See https://github.com/sussol/msupply/issues/8211.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #324.

## Description

This hurt to write. Logic is simple enough: if query is coming from mSupply, just hand it back what we know it wants. Goes without saying that at some point this will need to be fixed properly, or risk the `data-service` resolver logic becoming horribly hacky and difficult to work with!

Not going to add any tests for this, but will probably be a good idea to when we get a proper fix working (which will probably involve some updates to mSupply).

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [x] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

**Review by katherine-sussol — APPROVED** [review]
In conjunction with the next PR (#327) looks good, just one minor suggestion. 
I also noticed some of the codes are not 100% in line with the old API but that's totes a data issue and not code 🏃‍♀️ 

e.g. Amoxicillin & clavulanic acid 125mg &31.25mg /5mL oral liquid | 367734bf

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 1]
```diff
@@ -257,6 +319,47 @@ export class DgraphDataSource extends RESTDataSource {
...
+          product.children.forEach(route => {
+            if (route.children) {
+              route.children.forEach(doseForm => {
+                if (doseForm.children) {
+                  doseForm.children.forEach(doseQualifierOrDoseStrength => {
+                      if (doseQualifierOrDoseStrength.children) {
+                        doseQualifierOrDoseStrength.children.forEach(doseStrength => {
+                          const entity = {
+                            code: doseStrength.code,
+                            description: product.description + " " + doseQualifierOrDoseStrength.description + " " + doseStrength.description,
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts]: b67f8d8f04a2fd951a4b9330c69e36d7c6dae9fc.

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 2]
```diff
@@ -257,6 +319,47 @@ export class DgraphDataSource extends RESTDataSource {
...
+                            type: product.type,
+                          }
+                          mSupplyEntities.push(entity);
+                        })
+                      }
+                      else {
+                        const entity = {
+                          code: doseQualifierOrDoseStrength.code,
+                          description: product.description + " " + doseQualifierOrDoseStrength.description,
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts]: b67f8d8f04a2fd951a4b9330c69e36d7c6dae9fc.

**Comment by wlthomson** [comment]
@katherine-sussol @ujwalSussol mSupply screenshot demoing updating functionality:

image

**Pull request merged**